prov/efa/src/efa_hmem.c

```diff
@@ -386,6 +388,72 @@ int efa_hmem_info_initialize()
 	return ret;
 }
 
+/**
+ * @brief Copy data from a hmem device to a system buffer
+ *
+ * @param[in]    desc            Pointer to a memory registration descriptor
+ * @param[out]   dest            Destination system memory buffer
+ * @param[in]    src             Source hmem device memory
+ * @param[in]    size            Data size in bytes to copy
+ * @return       FI_SUCCESS status code on success, or an error code.
+ */
+int efa_copy_from_hmem(void *desc, void *dest, const void *src, size_t size)
```

Comment:
This is a fast path function, can we keep them static inline?

Reply:
Inlining works across TU boundaries with LTO. Even without LTO, this should have no effect wrt a compiler's inlining heuristics given how small the functions are. Furthermore, the callers themselves aren't `inline`d, so it's silly to draw the line at these 2 header-inlined functions and claim we're suddenly adding a new call boundary.

This was necessary because our include graph has become a tangled clusterfuck, and it's impossible to include tracepoints in these functions without moving them to a source file. Otherwise `#include efa_tp.h` in `efa_hmem.h` completes a circular include.